### PR TITLE
Allow resolved class to configure DNS settings

### PR DESCRIPTION
**Pull request by hfm**
ref: https://www.freedesktop.org/software/systemd/man/resolved.conf.html

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -12,6 +12,36 @@
 # @param resolved_ensure
 #   The state that the ``resolved`` service should be in
 #
+# @param $dns
```
Comment (bastelfreak) [inline manifests/init.pp]: there should probably be no $ in front of $dns

File: manifests/init.pp [hunk 2]
```diff
@@ -34,8 +64,16 @@
...
-  Enum['stopped','running']        $resolved_ensure  = 'running',
+  Boolean                                            $manage_resolved   = false,
+  Enum['stopped','running']                          $resolved_ensure   = 'running',
+  Optional[Variant[Array,String]]                    $dns               = undef,
+  Optional[Variant[Array,String]]                    $fallback_dns      = undef,
+  Optional[Variant[Array,String]]                    $domains           = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]         $llmnr             = undef,
+  Optional[Variant[Boolean,Enum['resolve']]]         $multicast_dns     = undef,
+  Optional[Variant[Boolean,Enum['allow-downgrade']]] $dnssec            = undef,
+  Optional[Boolean]                                  $cache             = undef,
```
Comment (bastelfreak) [inline manifests/init.pp]: In my opionion, everything that can be a boolean should default to false or true, so it doesn't need to be Optional.

**Comment by bastelfreak** [comment]
Hi @hfm, thanks for the PR! I will try to test them next week and review them properly.

File: manifests/init.pp [hunk 3]
```diff
@@ -34,8 +64,16 @@
 #   systemd-networkd take precedence over this setting. requires puppetlabs-inifile
 class systemd (
   Hash[String, Hash[String, Any]]  $service_limits   = {},
-  Boolean                          $manage_resolved  = false,
-  Enum['stopped','running']        $resolved_ensure  = 'running',
+  Boolean                                            $manage_resolved   = false,
```
Comment (bastelfreak) [inline manifests/init.pp]: please properly align the parameter at the `=`

File: manifests/resolved.pp [hunk 4]
```diff
@@ -1,12 +1,52 @@
...
+# @param cache
+#   Takes a boolean argument.
+#
+# @param dns_stub_listener
+#   Takes a boolean argument or one of "udp" and "tcp".
+#
 class systemd::resolved (
-  Enum['stopped','running'] $ensure = $systemd::resolved_ensure,
+  Enum['stopped','running'] $ensure                                = $systemd::resolved_ensure,
+  Optional[Variant[Array,String]] $dns                             = $systemd::dns,
```
Comment (bastelfreak) [inline manifests/resolved.pp]: Can you enforce the datatype in the Array? For example:

```puppet
Optional[Variant[Array[String],String]]
```

**Comment by bastelfreak** [comment]
Hi @hfm, can you please rebase against our current master branch and take a look at the inline comments?

**Comment by hfm** [comment]
@bastelfreak Hi, I (force) pushed. Please review it.

**Comment by bastelfreak** [comment]
@hfm can you check the failing spec tests?

**Comment by hfm** [comment]
Sorry. I fixed tests.

**Comment by bastelfreak** [comment]
Thanks for the great work @hfm ! Tested this locally and everything looks good.

**Comment by hfm** [comment]
Thanks!

**Pull request merged**